router/batchrouter/asyncdestinationmanager/yandexmetrica/yandexmetrica.go

```diff
@@ -0,0 +1,382 @@
+	// This client is used for uploading data to yandex metrica
+	yandexUploadManager.Client = oauthv2httpclient.NewOAuthHttpClient(
+		originalHttpClient,
+		oauthv2common.RudderFlowDelivery,
+		&cache,
+		backendConfig,
+		augmenter.GetAuthErrorCategoryForYandex,
+		optionalArgs,
+	)
+	return yandexUploadManager, nil
+}
+
+// return a success response for the poll request every time by default
+func (ym *YandexMetricaBulkUploader) Poll(pollInput common.AsyncPoll) common.PollStatusResponse {
+	return common.PollStatusResponse{}
+}
+
+// return a success response for the getUploadStats request every time by default
+func (ym *YandexMetricaBulkUploader) GetUploadStats(UploadStatsInput common.GetUploadStatsInput) common.GetUploadStatsResponse {
+	return common.GetUploadStatsResponse{}
+}
+
+func generateCSVFromJSON(jsonData []byte, goalId string) (string, string, error) {
+	// Define an empty map to store the parsed JSON data
+	var ymMsgs []YandexMetricaMessage
+	inputData := gjson.GetBytes(jsonData, "input").String()
+	err := json.Unmarshal([]byte(inputData), &ymMsgs)
+	if err != nil {
+		return "", "", fmt.Errorf("error while unmarshalling transformed response: %v", err)
+	}
+
+	ymMsgsBody := lo.Map(ymMsgs, func(ym YandexMetricaMessage, _ int) YandexMetricaMessageBody {
+		return ym.Message
+	})
+
+	// Open the CSV file for writing
+	localTmpDirName := fmt.Sprintf(`/%s/`, misc.RudderAsyncDestinationLogs)
+	tmpDirPath, err := misc.CreateTMPDIR()
+	if err != nil {
+		return "", "", fmt.Errorf("error creating tmp dir: %v", err)
+	}
+	folderPath := path.Join(tmpDirPath, localTmpDirName)
+	_, err = os.Stat(folderPath)
+	if os.IsNotExist(err) {
+		folderPath, _ = os.MkdirTemp(folderPath, "")
+	}
+	path := path.Join(folderPath, uuid.NewString())
+	csvFilePath := fmt.Sprintf(`%s.csv`, path)
+	csvFile, err := os.Create(csvFilePath)
+	if err != nil {
+		return "", "", fmt.Errorf("error creating csv file: %v", err)
+	}
+
+	defer csvFile.Close()
+
+	// Create a CSV writer
+	csvWriter := csv.NewWriter(csvFile)
+
+	// Define the header row based on key presence in "message" object
+	firstMsgIdDetails, err := ymMsgsBody[0].Id()
+	if err != nil {
+		return "", "", fmt.Errorf("missing 'ClientId', 'Yclid', or 'UserId' key in 'message' object")
+	}
+	idDecider := firstMsgIdDetails.headerName
+	header := []string{idDecider, "Target", "DateTime", "Price", "Currency"}
+	// Write the header row
+	err = csvWriter.Write(header)
+	if err != nil {
+		return "", "", fmt.Errorf("error writing header row: %v", err)
+	}
+
+	// Extract and write data rows
+	for index, ymMsg := range ymMsgsBody {
+		if ymMsg.Target == "" {
+			ymMsg.Target = goalId
+		}
+
+		idDetails, err := ymMsg.Id()
+		if err != nil {
+			continue
+		}
+		row := []string{
+			fmt.Sprintf("%v", idDetails.id),
+			fmt.Sprintf("%v", ymMsg.Target),
+			fmt.Sprintf("%v", ymMsg.DateTime),
+			fmt.Sprintf("%v", ymMsg.Price),
+			fmt.Sprintf("%v", ymMsg.Currency),
```

Comment:
what's the intended out here?  Is the behaviour of %v, match the expectations under all conditions? Should we use a more standarize way to encode them?